src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -14,153 +14,149 @@
+          '--service-job-number', '1',
+          '--commit-sha', commit_sha,
+          '--token', coveralls_token,
+        ]
+        cmd.extend(ordered_files)
+
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        (output, err) = p.communicate()
+
+        if p.returncode != 0:
+            raise Exception('Error while running grcov:\n' + err.decode('utf-8'))
+
+        return output
+
+    def clone_mozilla_central(self, revision):
+        shared_dir = self.repo_dir + '-shared'
+        cmd = hglib.util.cmdbuilder('robustcheckout',
+                                    'https://hg.mozilla.org/mozilla-central',
+                                    self.repo_dir,
+                                    purge=True,
+                                    sharebase=shared_dir,
+                                    branch=b'tip')
+
+        cmd.insert(0, hglib.HGPATH)
+        proc = hglib.util.popen(cmd)
+        out, err = proc.communicate()
+        if proc.returncode:
+            raise hglib.error.CommandError(cmd, proc.returncode, out, err)
+
+        hg = hglib.open(self.repo_dir)
+
+        hg.update(rev=revision, clean=True)
+
+    def run_command(self, cmd):
+        """
+        Run a command in the repo through subprocess
+        """
+        # Use gecko-env to run command
+        cmd = ['gecko-env', ] + cmd
+
+        # Run command with env
+        logger.info('Running repo command', cmd=' '.join(cmd))
+        proc = subprocess.Popen(cmd, cwd=self.repo_dir)
+        exit = proc.wait()
+
+        if exit != 0:
+            raise Exception('Invalid exit code for command {}: {}'.format(cmd, exit))  # NOQA
+
+    def build_files(self):
+        with open(os.path.join(self.repo_dir, '.mozconfig'), 'w') as f:
+            f.write('mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-firefox')
+
+        self.run_command(['./mach', 'configure'])
+        self.run_command(['./mach', 'build', 'pre-export'])
+        self.run_command(['./mach', 'build', 'export'])
+
+    def go(self):
+        task_id = taskcluster.get_last_task()
+
+        task_data = taskcluster.get_task_details(task_id)
+        revision = task_data['payload']['env']['GECKO_HEAD_REV']
+        logger.info('Revision %s' % revision)
+        commit_sha = self.get_github_commit(revision)
+        logger.info('GitHub revision %s' % commit_sha)
+
+        self.download_coverage_artifacts(task_id)
+
+        self.clone_mozilla_central(revision)
+        self.build_files()
+
+        output = self.generate_info(commit_sha, self.coveralls_token)
+
+        uploader.coveralls(output)
+        uploader.codecov(output, commit_sha, self.codecov_token)
+
+        coverage_by_dir.generate(self.repo_dir)
+
+    def __init__(self, cache_root, secrets, client_id=None, client_token=None):
+        assert os.path.isdir(cache_root), "Cache root {} is not a dir.".format(cache_root)
+        self.repo_dir = os.path.join(cache_root, 'mozilla-unified')
```

Comment:
We discussed on IRC, this task should not be affected by this problem (as we don't run more than one in parallel, at least for now).
This could be a problem for the static analysis or risk assessment tasks.